Libraries/LibWeb/URLPattern/URLPattern.cpp

```diff
@@ -51,11 +51,31 @@ WebIDL::ExceptionOr<GC::Ref<URLPattern>> URLPattern::create(JS::Realm& realm, UR
     return realm.create<URLPattern>(realm, pattern_or_error.release_value());
 }
 
+// https://urlpattern.spec.whatwg.org/#dom-urlpattern-test
+WebIDL::ExceptionOr<bool> URLPattern::test(URLPatternInput const& input, Optional<String> const& base_url) const
+{
+    // 1. Let result be the result of match given this's associated URL pattern, input, and baseURL if given.
+    auto result_or_error = m_url_pattern.match(input, base_url);
+    if (result_or_error.is_error())
+        return WebIDL::SimpleException { WebIDL::SimpleExceptionType::TypeError, result_or_error.error().message };
```

Comment:
If we find ourselves doing this manual error checking a lot, we could make `PatternErrorOr` an actual type with a custom `release_value` that constructs a `SimpleException`. Then you would be able to just `TRY` this.

(not something we need to do now, just an idea).

Reply:
I think I can't use `SimpleException` since it's defined in `LibWeb`, and I need this defined in `LibURL`. My idea for an alternative would be a `URL_PATTERN_TRY` macro to handle that boilerplate. Though, I think there will not be any more instances required of this - the LibWeb part of the specification is all upstreamed now. Remaining part is all the stuff needed for LibURL, where we're free to use `TRY` since it's just propagating an `ErrorOr`